engines/python/src/main/java/ai/djl/python/engine/Connection.java

```diff
@@ -120,10 +126,71 @@ CompletableFuture<Output> send(Input input) throws InterruptedException {
         return f;
     }
 
-    static String[] getPythonStartCmd(PyEnv pyEnv, Model model, int workerId, int port) {
+    static String[] getPythonStartCmd(
+            PyEnv pyEnv, Model model, int workerId, int port, int clusterSize) {
         Device device = model.getNDManager().getDevice();
         int deviceId = device.getDeviceId();
         int tensorParallelDegree = pyEnv.getTensorParallelDegree();
+        // int pipelineParallelDegree = pyEnv.getPipelineParallelDegree();
+        logger.info("Printing mpi boolean: {}", pyEnv.isMpiMode());
+
+        if (clusterSize > 1) {
+            String leaderAddress = Utils.getenv("LWS_LEADER_ADDRESS");
+            String lwsName = Utils.getenv("LWS_NAME");
+            String namespace = Utils.getenv("NAMESPACE");
+            String groupIndex = Utils.getenv("GROUP_INDEX");
+            String[] hosts = new String[clusterSize];
+            hosts[0] = leaderAddress;
+            for (int i = 1; i < clusterSize; i++) {
+                hosts[i] =
+                        String.format("%s-%s-%d.%s.%s", lwsName, groupIndex, i, lwsName, namespace);
+            }
+
+            String cudaDevices = getVisibleDevices(workerId, tensorParallelDegree);
+            logger.info("Set before mpirun CUDA_VISIBLE_DEVICES={}", cudaDevices);
+            String[] args = new String[36];
+            args[0] = "mpirun";
+            args[1] = "-np";
+            // TODO: When we support multi nodes, change it to the product of tensor parallel value
+            // and
+            // pipeline parallel value.
+            args[2] = String.valueOf(clusterSize * tensorParallelDegree);
+            args[3] = "--host";
+            args[4] = String.join(",", hosts);
+            args[5] = "--allow-run-as-root";
+            args[6] = "--bind-to";
+            args[7] = "none";
+            args[8] = "--mca";
+            args[9] = "btl_vader_single_copy_mechanism";
+            args[10] = "none";
+            args[11] = "--tag-output";
+            args[12] = "-x";
+            args[13] = "FI_PROVIDER=efa";
+            args[14] = "-x";
+            args[15] = "RDMAV_FORK_SAFE=1";
+            args[16] = "-x";
+            args[17] = "FI_EFA_USE_DEVICE_RDMA=1";
+            args[18] = "-x";
+            args[19] = "LD_LIBRARY_PATH";
+            args[20] = "-x";
+            args[21] = "PYTHONPATH";
+            args[22] = "-x";
+            args[23] = "CUDA_VISIBLE_DEVICES=" + cudaDevices;
+            args[24] = pyEnv.getPythonExecutable();
+            args[25] = PyEnv.getEngineCacheDir() + "/djl_python_engine.py";
+            args[26] = "--model-dir";
+            args[27] = model.getModelPath().toAbsolutePath().toString();
+            args[28] = "--entry-point";
+            args[29] = pyEnv.getEntryPoint();
+            args[30] = "--sock-type";
```

Comment:
We should pass `--sock_name 0.0.0.0` to python

Reply:
But the socket name 0.0.0.0 doesn't work for tcp.